sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -131,13 +131,19 @@ def _csr_mean_variance_axis0(np.ndarray[floating, ndim=1, mode="c"] X_data,
         np.ndarray[floating, ndim=1] sum_weights_nz = \
             np.zeros(shape=n_features, dtype=dtype)
 
+        np.ndarray[np.uint64_t, ndim=1] counts = np.full(
```

Comment:
A more explicit name:

```suggestion
        np.ndarray[np.uint64_t, ndim=1] counts_nan = np.full(
```

Reply:
`sum_weights` is the sum of all weights where X is not nan
`sum_weights_nan` is the sum of weights where X is nan
`sum_weights_nz` is the sum of weights where X is non zero

Following the same scheme:
`counts` is the number of elements which are not nan
`counts_nz` is the number of elements which are non zero

I'd rather keep that. Maybe you missed that the increment is negative (`counts[col_ind] -= 1`).
Let me try to reorder the code such that the match is clearer (and remove `sum_weights_nan`, we actually need only 2 out the 3 arrays).

Reply:
I was thinking about the negation of it, so it the suggestion should have been `counts_non_nan`.

If we do not change the name, I think we should at least put a comment above `counts` to say that it is the number of elements which are not nan.

Reply:
I added comments to describe the different arrays